app/Importers/VeoDataImporter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Importers;
+
+use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\DomCrawler\Crawler;
+
+class VeoDataImporter extends DataImporter
+{
+    private const COUNTRY_NAME = "United States";
+
+    protected Crawler $sections;
+
+    public function extract(): static
+    {
+        try {
+            $response = $this->client->get("https://www.veoride.com/locations/");
+            $html = $response->getBody()->getContents();
+        } catch (GuzzleException) {
+            $this->createImportInfoDetails("400", self::getProviderName());
+
+            $this->stopExecution = true;
+
+            return $this;
+        }
+
+        $crawler = new Crawler($html);
+        $this->sections = $crawler->filter("body > div > div .elementor-226069 > div .elementor-row");
+
+        if (count($this->sections) === 0) {
+            $this->createImportInfoDetails("204", self::getProviderName());
+            $this->stopExecution = true;
+        }
+
+        return $this;
+    }
+
+    public function transform(): void
+    {
+        if ($this->stopExecution) {
+            return;
+        }
+
+        $existingCityProviders = [];
+
+        foreach ($this->sections as $section) {
+            if ($section->nodeName === "div") {
+                foreach ($section->childNodes as $city) {
+                    $usaStates = [
+                        "Alabama", "Alaska", "Arizona", "Arkansas", "California",
+                        "Colorado", "Connecticut", "Delaware", "Florida", "Georgia",
+                        "Hawaii", "Idaho", "Illinois", "Indiana", "Iowa", "Kansas",
+                        "Kentucky", "Louisiana", "Maine", "Maryland", "Massachusetts",
+                        "Michigan", "Minnesota", "Mississippi", "Missouri", "Montana",
+                        "Nebraska", "Nevada", "New Hampshire", "New Jersey", "New Mexico",
+                        "New York", "North Carolina", "North Dakota", "Ohio", "Oklahoma",
+                        "Oregon", "Pennsylvania", "Rhode Island", "South Carolina",
+                        "South Dakota", "Tennessee", "Texas", "Utah", "Vermont",
+                        "Virginia", "Washington, DC", "West Virginia", "Wisconsin", "Wyoming",
+                    ];
```

Comment:
Please move this to some protected method. It will be more readable.

Reply:
changed

Reply:
Add `array` type to the property.

Reply:
added